sampling/rate_limited/lib/opentelemetry/sampling/rate_limited/sampler.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Sampling
+    module RateLimited
+      class Sampler
+        def initialize(credits_per_second)
+          @credits_per_second = credits_per_second
+          @balance = 0
+          @last_tick = Time.now.to_i
```

Comment:
Please use Process clocks for more precise time calculations:

https://ruby-doc.org/3.2.2/Process.html#method-c-clock_gettime